lib/browser/guest-window-manager.js

```diff
@@ -131,14 +147,7 @@ const createGuest = function (embedder, url, referrer, frameName, options, postD
     }
     if (postData != null) {
       loadOptions.postData = postData
-      loadOptions.extraHeaders = 'content-type: application/x-www-form-urlencoded'
-      if (postData.length > 0) {
-        const postDataFront = postData[0].bytes.toString()
-        const boundary = /^--.*[^-\r\n]/.exec(postDataFront)
-        if (boundary != null) {
-          loadOptions.extraHeaders = `content-type: multipart/form-data; boundary=${boundary[0].substr(2)}`
-        }
-      }
+      loadOptions.extraHeaders = headersForPostData(postData)
```

Comment:
is 'postData` guaranteed to be a buffer here? because the docs suggest it could be any of a number of different things.

Reply:
Ugh, no, it's not at all guaranteed 😕

Reply:
Actually, good news, I read the code wrong. Added comments to make it clear what is happening here, but this is safe in my tests of target="_blank" window openings.